File: src/main/java/com/tartner/dancehours/web/donations/DonationEntryQueryRepository.java
package com.tartner.dancehours.web.donations;

import com.tartner.utilities.UUIDGenerator;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.UUID;

@Service
public class DonationEntryQueryRepository {
    @Autowired private UUIDGenerator uuidGenerator;

    public DonationEntryForm CreateNewEntryForm( String userId ) {
        DonationEntryForm form = new DonationEntryForm();
        form.setEntryId( uuidGenerator.newId() );
        form.setMemberDonationWasProvidedFor(
            defaultMemberIdForDonations( userId ) );
        return form;
    }

    private UUID defaultMemberIdForDonations( final String userId ) {
        // TODO: change default implementation
        return null;
    }
}

File: src/main/java/com/tartner/dancehours/web/donations/DonationEntryQueryRepository.kt
package com.tartner.dancehours.web.donations

import com.tartner.dancehours.DanceHoursId
import com.tartner.utilities.IdentifierGenerator
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
public class DonationEntryQueryRepository @Autowired constructor( val identifierGenerator : IdentifierGenerator) {

    public fun CreateNewEntryForm(userId: String): DonationEntryForm {
        val form = DonationEntryForm().apply {
            entryId = identifierGenerator.newId()
            memberDonationWasProvidedFor = defaultMemberIdForDonations(userId)
            donationTypeId = defaultDonationType(userId)
        }

        return form
    }

    // TODO: make this work for real
    private fun defaultDonationType(userId: String): DanceHoursId {
        return identifierGenerator.newId()
    }

    private fun defaultMemberIdForDonations(userId: String): DanceHoursId {
        return identifierGenerator.newId()
    }
}
